File: src/main/java/com/telenordigital/prime/config/EventProcessorConfiguration.java
package com.telenordigital.prime.config;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.hibernate.validator.constraints.NotEmpty;

public final class EventProcessorConfiguration {
    @NotEmpty
    @JsonProperty("databaseName")
    private String databaseName;

    @NotEmpty
    @JsonProperty("configFile")
    private String configFile;

    public String getDatabaseName() {
        return databaseName;
    }

    public String getConfigFile() {
        return configFile;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/config/EventProcessorConfiguration.kt
package org.ostelco.prime.config

import com.fasterxml.jackson.annotation.JsonProperty
import org.hibernate.validator.constraints.NotEmpty

class EventProcessorConfiguration {

    @NotEmpty
    @JsonProperty("configFile")
    lateinit var configFile: String

    @NotEmpty
    @JsonProperty("projectId")
    lateinit var projectId: String

    @NotEmpty
    @JsonProperty("topicId")
    lateinit var topicId: String
}
